src/cljs/ataru/hakija/application_handlers.cljs

```diff
@@ -170,19 +171,61 @@
             :url     (str "/hakemus/auth/session")
             :handler [:application/handle-oppija-session-login-refresh]}}))
 
+(reg-event-db
+  :application/show-session-expiry-warning-dialog
+  [check-schema-interceptor]
+  (fn [db [_ warning-minutes]]
+    (js/console.log (str "Setting active warning for... " warning-minutes))
+    (assoc-in db [:oppija-session :session-expires-in-minutes-warning] warning-minutes)))
+
+(reg-event-fx
+  :application/show-session-expiry-warning-dialog-if-needed
+  [check-schema-interceptor]
+  (fn [{:keys [db]} [_ seconds-left-in-session]]
+    (let [polling-interval-seconds  (/ (-> js/config
+                                           js->clj
+                                           (get "oppija-session-polling-interval"))
+                                       1000)
+          warning-target-minutes '(30 20 10)
+          previous-warnings (get-in db [:oppija-session :activated-warnings] #{})
+          extra-margin-seconds 30;ettei käy ikäviä harvinaisia yllätyksiä esim. hitaiden kutsujen kanssa. Tästä ei haittaa, pidentää vain vähän setTimeoutille passattavaa odotusaikaa.
+          should-set-warning-fn (fn [warning-target seconds-left]
```

Comment:
Should kuulostaa vähän testikoodilta, voisiko olla toisen niminen esim.. get-warning-value-to-set-fn?

Reply:
Mietin myös et tän sisäfunkkarin voisi olla sopiva eriyttää ja tehdä sille yksikkötestit?